core/src/main/java/bio/terra/pearl/core/service/kit/KitRequestService.java

```diff
@@ -57,43 +60,40 @@ public KitRequestService(KitRequestDao dao,
-     *
-     * These steps are all performed in a single transaction. However, each could fail independently. If there's an
-     * error from Pepper, then the sample kit request will not be saved in Juniper, and it's safe to assume that there
-     * is no new request in Pepper. If there's an error saving the status response from Pepper, then it's possible for
-     * there to be a request in Pepper that does not exist in Juniper. We could potentially split that step into its own
-     * transaction to help avoid that potential inconsistency.
+     * Throws PepperApiException if the Pepper API request failed
      */
-    @Transactional
-    public KitRequest requestKit(AdminUser adminUser, String studyShortcode, Enrollee enrollee, String kitTypeName) {
+    public KitRequest requestKit(AdminUser adminUser, String studyShortcode, Enrollee enrollee, String kitTypeName)
+    throws PepperApiException {
         // create and save kit request
-        Profile profile = profileService.loadWithMailingAddress(enrollee.getProfileId())
-                .orElseThrow(() -> new RuntimeException("Missing profile for enrollee: " + enrollee.getShortcode()));
+        if (enrollee.getProfileId() == null) {
+            throw new IllegalArgumentException("No profile for enrollee: " + enrollee.getShortcode());
+        }
+        Profile profile = profileService.loadWithMailingAddress(enrollee.getProfileId()).get();
         PepperKitAddress pepperKitAddress = makePepperKitAddress(profile);
-        var kitRequest = createKitRequest(adminUser, enrollee, pepperKitAddress, kitTypeName);
+        KitRequest kitRequest = assemble(adminUser, enrollee, pepperKitAddress, kitTypeName);
 
         // send kit request to DSM
-        var response = pepperDSMClient.sendKitRequest(studyShortcode, enrollee, kitRequest, pepperKitAddress);
-
-        // save DSM response/status with Juniper KitRequest
         try {
-            var untypedStatuses = PepperKitStatusResponse.extractUntypedKitStatuses(response, objectMapper);
-            // Pepper response from requesting it kit must have one and only one kit status
-            var pepperStatusJson = objectMapper.writeValueAsString(untypedStatuses.get(0));
+            PepperKitStatus dsmKitStatus = pepperDSMClient.sendKitRequest(studyShortcode, enrollee, kitRequest, pepperKitAddress);
+            // write out the PepperKitStatus as a string for storage
+            var pepperStatusJson = objectMapper.writeValueAsString(dsmKitStatus);
             kitRequest.setDsmStatus(pepperStatusJson);
-            saveKitStatus(kitRequest, pepperStatusJson, kitRequest.getCreatedAt());
+        } catch (PepperParseException e) {
+            // response was successful, but we got unexpected format back from pepper
+            // we want to log the error, but still continue on to saving the kit
+            log.error("Unable to parse kit response status from Pepper: kit id %s", kitRequest.getId());
```

Comment:
we're very leery of including exceptions and responses in logs--the directive from AppSec/Compliance is to avoid logging PII/PHI, and logging the exception/response here seems like a reasonable risk that the response from Pepper would have PII/PHI in it

Reply:
Not sure I agree with that.